R/Registry.R

```diff
@@ -346,3 +369,75 @@ Registry <- R6::R6Class( # nolint object_name_linter
     }
   )
 )
+
+create_record_from_python <- function(py_record, instance) {
+
+  py_classes <- class(py_record)
+
+  # Skip related fields for now
+  if ("django.db.models.manager.Manager" %in% py_classes) {
+    return(NULL)
+  }
+
+  class_split <- strsplit(py_classes[1], "\\.")[[1]]
+  module_name <- class_split[1]
+  if (module_name == "lnschema_core") {
+    module_name <- "core"
+  }
+  registry_name <- tolower(class_split[3])
+
+  registry <- instance$get_module(module_name)$get_registry(registry_name)
+  fields <- registry$get_field_names()
+
+  record_list <- map(fields, function(.field) {
+    value <- tryCatch(
+      py_record[[.field]],
+      error = function(err) {
+        NULL
+      }
+    )
+    if (inherits(value, "lnschema_core.models.Record")) {
+      value <- create_record_from_python(value, instance)
+    }
+    value
+  }) |>
+    setNames(fields)
+
+  record_class <- registry$get_record_class()
+  temp_record_class <- create_temporary_record_class(record_class)
```

Comment:
Do we need to run `create_temporary_record_class` every time we create a record?

Or could it also be added to the registry (e.g. `registry$get_transient_record_class()`)

Reply:
Yes, if we are happy to add a bunch of extra classes that will rarely be used. Maybe we only add them to the default instance?

Reply:
are there that many classes which might need to be generated?

alternatively, `registry$get_record_class()` could have a few optional arguments to choose whether the record class is transient or not. This in turn could use lazy loading to ensure that we don't generate the same classes over and over again